source/common/init/manager_impl.cc

```diff
@@ -18,13 +21,15 @@ void ManagerImpl::add(const Target& target) {
   case State::Uninitialized:
     // If the manager isn't initialized yet, save the target handle to be initialized later.
     ENVOY_LOG(debug, "added {} to {}", target.name(), name_);
+    ++target_names_count_[target.name()];
```

Comment:
This should be before `swtich`right below (`++count_;`)

Reply:
I suppose we should not add target into the hash table when the state is ```Initialized```.

Reply:
Right, the statement after `case State::Initialized` is `ASSERT(false)` so it will crash the whole process anyway :)

Reply:
Yeah, updated